security-oauth2/src/main/java/io/micronaut/security/oauth2/configuration/OauthClientConfigurationProperties.java

```diff
@@ -435,6 +440,16 @@ public String getHeaderName() {
             }
 
         }
+
+        @Override
+        @NonNull
+        public Map<String, String> getAdditionalRequestParams() {
+            return additonalRequestParams;
+        }
+
+        public void setAdditionalRequestParams(@MapFormat(transformation = MapFormat.MapTransformation.FLAT) Map<String, String> additionalRequestParams) {
```

Comment:
It misses javadoc. 

Configuration Reference is populated with javadoc of ConfigurationProperites setters. Thus, it is important to provide good Javadoc here. 

Run `./gradlew docs` and you will see the configuration reference documentation being generated.

Reply:
Javadoc added, please let me know if there's anything else that should be added.

Should there be an example for the Auth0 use-case maybe?

Reply:
> Should there be an example for the Auth0 use-case maybe?
 

We will change Auth0 Client Credentials guide to use this feature once micronaut security 3.1 is included in the Micronaut bom core

 https://github.com/micronaut-projects/micronaut-guides/pull/513